native/acter/src/api/room.rs

```diff
@@ -1445,22 +1451,44 @@ impl Room {
                 let event_content = evt.event.deserialize_as::<RoomMessageEvent>()?;
                 let original = event_content
                     .as_original()
-                    .context("Couldn't get original msg")?;
-                if !matches!(
-                    &original.content.msgtype,
-                    MessageType::Image(_)
-                        | MessageType::Audio(_)
-                        | MessageType::Video(_)
-                        | MessageType::File(_)
-                ) {
-                    bail!("This message type is not downloadable");
+                    .expect("Couldn't get original msg");
+                if is_thumb {
+                    let available = matches!(
+                        &original.content.msgtype,
+                        MessageType::Image(_)
+                            | MessageType::Video(_)
+                            | MessageType::File(_)
+                            | MessageType::Location(_)
+                    );
+                    if !available {
+                        bail!("This message type is not downloadable");
+                    }
```

Comment:
That's an odd way of writing this. Can't we directly do the if please?
```suggestion
                    if !matches!(
                        &original.content.msgtype,
                        MessageType::Image(_)
                            | MessageType::Video(_)
                            | MessageType::File(_)
                            | MessageType::Location(_)
                    ){
                        bail!("This message type is not downloadable");
                    }
```

Reply:
When `if` case becomes simple, github workflow requires `else` case to be changed into `else if`.
It looks like the following code snippet.
```rs
                if is_thumb
                    && !matches!(
                        &original.content.msgtype,
                        MessageType::Image(_)
                            | MessageType::Video(_)
                            | MessageType::File(_)
                            | MessageType::Location(_)
                    ) {
                    bail!("This message type is not downloadable");
                } else if !matches!(
                    &original.content.msgtype,
                    MessageType::Image(_)
                        | MessageType::Audio(_)
                        | MessageType::Video(_)
                        | MessageType::File(_)
                ) {
                    bail!("This message type is not downloadable");
                }
```
As you see, 1st block seems to have check of `is_thumb` more than 2nd block.
I think this code may let developer misunderstand easily.

Reply:
So I placed the variable of `available` in order to give developer clear understanding and block failure in github workflow lint.